dao/tenant_dao.go

```diff
@@ -70,3 +77,141 @@ func (t *tenantDaoImpl) TenantByIdentity(id *identity.Identity) (*m.Tenant, erro
+	var translatedTenants, untranslatedTenants uint64
+	// Keep track of each of the tenant translations.
+	var tenantTranslations []m.TenantTranslation
+
+	// Run everything inside a transaction to avoid inconsistencies. This way if new tenants are added/removed while
+	// the translation is in progress, all the counts and the results will still match.
+	err := DB.Debug().Transaction(func(tx *gorm.DB) error {
+		// Get the count to be able to process the tenants in batches of 100 tenants.
+		err := tx.
+			Model(&m.Tenant{}).
+			Where(untranslatedTenantsWhereCondition).
+			Count(&translatableTenants).
+			Error
+
+		if err != nil {
+			return fmt.Errorf("unable to count untranslated tenants: %w", err)
+		}
+
+		translator := tenantid.NewTranslator(config.Get().TenantTranslatorUrl)
+		for i := int64(0); i < translatableTenants; i += 100 {
+			// Grab all the EBS account numbers to be processed.
+			var ebsAccountNumbers []string
+			err := tx.
+				Model(&m.Tenant{}).
+				Where(untranslatedTenantsWhereCondition).
+				Pluck("external_tenant", &ebsAccountNumbers).
+				Offset(int(i)).
+				Limit(100).
+				Error
+
+			if err != nil {
+				return fmt.Errorf("unable to fetch the tenants to translate them: %w", err)
+			}
+
+			// Batch process 100 EBS account numbers.
+			results, err := translator.EANsToOrgIDs(context.Background(), ebsAccountNumbers)
+			if err != nil {
+				return fmt.Errorf("error when calling the translation service: %w", err)
+			}
+
+			for _, result := range results {
+				var tenantTranslation = m.TenantTranslation{}
+
+				// Did the translation service return an error?
+				if result.Err != nil {
+					logging.Log.WithField("external_tenant", *result.EAN).Errorf(`could not translate to "org_id": %s`, err)
+
+					tenantTranslation.ExternalTenant = *result.EAN
+					tenantTranslation.Error = err
+					tenantTranslations = append(tenantTranslations, tenantTranslation)
+
+					untranslatedTenants++
+					continue
+				}
+
+				// Grab the updated tenant for the "translated tenant" struct.
+				var tenant m.Tenant
+				// Update the tenant with the translated "orgId".
+				dbResult := tx.
+					Model(&tenant).
+					Clauses(clause.Returning{Columns: []clause.Column{{Name: "id"}}}).
+					Where("external_tenant = ?", result.EAN).
+					Updates(map[string]interface{}{
+						"org_id": result.OrgID,
+					})
+
+				// If the update yielded 0 affected rows that means that no tenant could be found with that
+				// "external_tenant".
+				if dbResult.RowsAffected == 0 {
+					translationError := errors.New(`could not translate to "org_id": external tenant not found`)
+
+					logging.Log.WithField("external_tenant", *result.EAN).Error(translationError)
+
+					tenantTranslation.ExternalTenant = *result.EAN
+					tenantTranslation.Error = translationError
+					tenantTranslations = append(tenantTranslations, tenantTranslation)
+
+					untranslatedTenants++
+					continue
+				}
+
+				if err != nil {
```

Comment:
it looks that this `err != nil` is covered on 134 or am I missing something ?

Reply:
You're right. Actually this was meant to be `dbResult.Error`, to be able to log a potential error with the update statement.